### ISSUE TITLE
[Enchancement]: Need a proxy configuration

### ISSUE THREAD

**Issue by hothoney** [maxijonson/gpt-turbo#2]
### Type

New feature

### Description

gpt service is not available in my country, so I have to use a proxy to reach it

### Benefit

as above

### Participation

None

**Comment by maxijonson** [comment]
Hi! Thanks for raising the issue. I've never worked with proxies in libraries, but I do understand the need for them in different situations. 

I'll look into what can be done about this!

**Comment by maxijonson** [comment]
There seem to be a related issue on OpenAI's Node API with a potential fix through configuring the axios instance. I'll try to implement some config options to the Conversation object to do this. However, I'm unsure how to test this locally, but I'll try to figure something out!

https://github.com/openai/openai-node/issues/85

**Comment by hothoney** [comment]
Another related issue is https://github.com/openai/openai-node/issues/92.
This [solution](https://community.openai.com/t/4039error-but-functional-on-postman-copy-postman-code-and-paste-it-on-project-dont-work/84299) can resolve this issue, hope it can help

**Comment by maxijonson** [comment]
Hey! Sorry it took me a while to ship this, I was working on the message streaming feature at the time the issue was raised. Thanks for your patience!

I implemented a way to pass an [`AxiosRequestConfig`](https://axios-http.com/docs/req_config) object to both the `Conversation` constructor and its `prompt` method.

This is available on version `2.1.0`.

```ts
import { Conversation } from 'gpt-turbo';

(async () => {
    const conversation = new Conversation(
        {
        apiKey: /* Your OpenAI API key */,
        }, 
        {
            proxy: {
                protocol: /* Your proxy protocol */,
                host: /* Your proxy host */,
                port: /* Your proxy port */,
                auth: {
                    username: /* Your proxy username */,
                    password: /* Your proxy password */,
                },
            }
        }
    );

    const response = await conversation.prompt(
        "What is TypeScript?",
        {
            proxy: {
                protocol: /* Your proxy protocol */,
                host: /* Your proxy host */,
                port: /* Your proxy port */,
                auth: {
                    username: /* Your proxy username */,
                    password: /* Your proxy password */,
                },
            }
        }
    );
    console.log(`Response: ${response.content}`);
})();
```

Let me know if you have any questions or other issues regarding the proxy. I haven't been able to test it, since I've never used such features, but it should work since there's no implementation on gpt-turbo apart from passing your options straight to OpenAI's SDK. Thanks again for your issue!
